Tests/SpeechToTextV1Tests/SpeechToTextTests.swift

```diff
@@ -73,6 +89,196 @@ class SpeechToTextTests: XCTestCase {
+        let failToCreateCustomization =  { (error: Error) in
+            XCTFail("Failed to create the new customization: \(error)")
+            return
+        }
+        
+        let description = "Create a new customization."
+        let expectation = self.expectation(description: description)
+        speechToText.createCustomization(
+            withName: self.trainedCustomizationName,
+            withBaseModelName: "en-US_BroadbandModel",
+            description: self.trainedCustomizationDescription,
+            failure: failToCreateCustomization) { customization in
+                
+            self.trainedCustomizationID = customization.customizationID
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    func addCorpusToTrainedCustomization() {
+        
+        let failToCreateCorpus =  { (error: Error) in
+            XCTFail("Failed to create the new corpus: \(error)")
+            return
+        }
+        
+        guard let corpusFile = loadFile(name: "healthcare", withExtension: "txt") else {
+            XCTFail("Failed to load file needed to create the corpus.")
+            return
+        }
+        
+        let description = "Add a corpus for the customization."
+        let expectation = self.expectation(description: description)
+        speechToText.addCorpus(
+            withName: corpusName,
+            fromFile: corpusFile,
+            customizationID: self.trainedCustomizationID,
+            failure: failToCreateCorpus) {
+                
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    func trainCustomizationWithCorpus() {
+        
+        let failToGetCorpusStatus = { (error: Error) in
+            XCTFail("Failed to get the status of the new corpus: \(error)")
+            return
+        }
+        
+        let failToTrainCustomization = { (error: Error) in
+            XCTFail("Failed to train the new customization: \(error)")
+            return
+        }
+        
+        let failToGetCustomizationStatus = { (error: Error) in
+            XCTFail("Failed to get the status of the new customization: \(error)")
+            return
+        }
+        
+        var trained = false
+        var tries = 0
+        while(!trained) {
+            tries += 1
+            let description = "Wait until the corpus is processed before training the customization."
+            let expectation = self.expectation(description: description)
+            
+            speechToText.getCorpus(
+                withName: corpusName,
+                customizationID: self.trainedCustomizationID,
+                failure: failToGetCorpusStatus) { corpus in
+                    
+                    if corpus.status == .analyzed {
+                        trained = true
+                    }
+                    expectation.fulfill()
+            }
+            waitForExpectations()
+            
+            if tries > 10 {
```

Comment:
Hmm to be honest, I randomly chose 10 as the "max number of tries" (good point about pulling that out and making it a variable, btw - I'll go ahead and pull it out as a variable).

10 seemed like a good number, because training takes awhile sometimes. Trying 5 times wasn't quite enough - it'd fail pretty often with only 5 tries. This check is needed because all I'm checking for is if the corpus is processed - if there's an error, the corpus will never have a status of `processed`, and we would be stuck in an infinite loop here.